crates/harp/src/parser/srcref.rs

```diff
@@ -118,11 +118,10 @@ impl TryFrom<RObject> for SrcRef {
 /// Creates the same sort of srcfile object as with `parse(text = )`.
 /// Takes code as an R string containing newlines, or as a R vector of lines.
 impl SrcFile {
-    pub fn new_virtual(text: &str) -> harp::Result<Self> {
-        let input = crate::as_parse_text(text);
+    pub fn new_virtual(text: RObject) -> harp::Result<Self> {
```

Comment:
```suggestion
    fn new_virtual(text: RObject) -> harp::Result<Self> {
```

It feels like you'll want to make this with `try_from()` now? Even if you are coming from an `RObject` (which we don't have a `try_from()` method for, but we could.

Reply:
hmm I'm not sure, it feels strange not to have "official" constructors. But I'll make this ctor private.